Lib/enum.py

```diff
@@ -414,6 +529,39 @@ def __new__(metacls, cls, bases, classdict, **kwds):
             if _order_ != enum_class._member_names_:
                 raise TypeError('member order does not match _order_')
         #
+        # remove Flag structures if final class is not a Flag
+        if (
+                Flag is None and cls != 'Flag'
+                or Flag is not None and not issubclass(enum_class, Flag)
+            ):
+            delattr(enum_class, '_boundary_')
+            delattr(enum_class, '_flag_mask_')
+            delattr(enum_class, '_all_bits_')
+            delattr(enum_class, '_inverted_')
+        elif Flag is not None and issubclass(enum_class, Flag):
+            # ensure _all_bits_ is correct and there are no missing flags
+            single_bit_total = 0
+            multi_bit_total = 0
+            for flag in enum_class._member_map_.values():
+                if _is_single_bit(flag._value_):
+                    single_bit_total |= flag._value_
+                else:
+                    # multi-bit flags are considered aliases
+                    multi_bit_total |= flag._value_
+            if enum_class._boundary_ is not KEEP:
+                missed_bits = multi_bit_total & ~single_bit_total
+                missed = []
+                while missed_bits:
+                    i = 2 ** (missed_bits.bit_length() - 1)
+                    missed.append(i)
+                    missed_bits &= ~i
```

Comment:
The other thing to say here, is that `bit_length()` is being used to walk the bits from MSB to LSB order.  I suspect it's significantly less efficient than the LSB to MSB walking of the Kernighan algorithm, because `bit_length()` probably has some math log calls, etc.  I'd like to try some `timeit` tests to verify.

This is why I'm suggesting that LSB to MSB be used in the public iterator, repr, etc.

Reply:
walking from LSB to MSB is 2x faster

```python3
def bits_lsb(x):
    while x:
        b = x & (~x + 1)
        yield b
        x ^= b

def bits_msb(x):
    while x:
        b = 2 ** (x.bit_length() - 1)
        yield b
        x &= ~b

>>> timeit.timeit('list(bits_lsb(65535))', globals=vars())
4.1443982159998995
>>> timeit.timeit('list(bits_msb(65535))', globals=vars())
9.664363892000438
```

Reply:
Done.